engine/schema/src/main/java/com/cloud/storage/dao/VolumeDaoImpl.java

```diff
@@ -811,4 +820,16 @@ public List<VolumeVO> listByIds(List<Long> ids) {
         sc.setParameters("idIN", ids.toArray());
         return listBy(sc, null);
     }
+
+    @Override
+    public VolumeVO persist(VolumeVO entity) {
+        Object obj = CallContext.current().getContextParameter(String.format("%s-%s", ResourceReservation.class.getSimpleName(), Resource.ResourceType.volume.getName()));
+        if (obj instanceof Long) {
+            long reservationId = (long) obj;
+            if (reservationDao.findById(reservationId) != null) {
+                entity.setResourceReservationId((long) obj);
+            }
```

Comment:
so what if we have a reservation id but it does not exist in the DB? (unit test worthy?)